### DISCUSSION TITLE
Health Check Endpoint Warnings (MS8)

### DISCUSSION THREAD

**Discussion by ghost** [eclipse-edc/Connector#2583] [Q&A]
Hi,

We want to use health checks in our docker containers. 

Currently there are two healthcheck endpoints, we are using the endpoint under the management context as suggested by the warning introduced in PR [1]. However we are currently getting a warning every 5 seconds about an outdated health URL..

Which is the correct health endpoint? If its under the mangegement API, why is it under the management API?
Endpoints under the management API might be secured by different means, which might make the management API unsuitable for Docker health check endpoints.

We are not using the default ports and paths:
`http://localhost:11004/api/v1/management/check/health` (health check in management-context)
`http://localhost:8181/api/check/health` (health check in web-context)

Both endpoints keep printing the same warning message:
`The check/* endpoint has been moved under the 'management' context, please update your url accordingly, because this endpoint will be deleted in the next releases`

[1] https://github.com/eclipse-edc/Connector/pull/2217/files

**Comment by ndr-brt** [comment]
The idea was to define 4 clear api contexts, as stated in [this decision record](https://github.com/eclipse-edc/Connector/blob/3159a676ab49c32738b66c4283784d8ad5f69475/docs/developer/decision-records/2022-11-09-api-refactoring/renaming.md).

I'd say that in fact for most of the cases would be preferred to have the health check exposed on a different api context (like the `default` one). Do do that it is just enough to register the `ObservabilityApiController` on a specific context in an extension.

If we want to do this upstream let's wait for more opinions, then an issue/PR would be appreciated. 
In the meantime you can use the workaround I explained.

**Reply by florianrusch-zf** [reply]
+1 for putting it back into a separate context, which of course could also been secured with Api-Key, BasicAuth or what ever.

One use case/example which requires this would be:
In kubernetes you can configure prometheus (with the custom resource `ServiceMonitor`) to grab metrics from a specific port from all pods in a namespace. If you have the metrics within the management api which is secured by e.g. an api key, you would have to configure prometheus to scrap the metrics from each pod individually (one `ServiceMonitor` resource for each deployed EDC instance) .

**Reply by domreuter** [reply]
+1 for putting the check endpoints back into an context without authentication.

We are using the endpoints for our readiness, startup and liveness probes in K8s. In general it is also possible to do these calls with a prober authentication (Basic, ApiKey) but we are actually using an individual OAuth on the management context :-)

**Reply by ndr-brt** [reply]
issue/PR will be appreciated

**Reply by SebastianOpriel** [reply]
+1 for @domreuter 's suggestion

**Reply by ndr-brt** [reply]
just for the record, in https://github.com/eclipse-tractusx/tractusx-edc/tree/main/edc-extensions/observability-api-customization there's an extension that satisfies this need

**Reply by SebastianOpriel** [reply]
@ndr-brt 
This is now to be found here, right?
https://github.com/eclipse-edc/Connector/tree/main/extensions/common/api/api-observability
